spec/services/admin/flags/notify_user_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe Admin::Flags::NotifyUser do
+  subject(:service) { described_class.call(admin: admin, flag: flag, notification: notification) }
+
+  let(:admin) { create(:user, admin: true) }
+  let(:flag) { create(:flag, project_submission: project_submission) }
+  let(:notification) { double('Notifications::DeadLink', call: 'called') }
+  let(:project_submission) { create(:project_submission, user: user) }
+  let(:user) { create(:user) }
+
+  describe '#call' do
+    it 'sets the flags taken action to notified_user' do
+      expect { service }.to change { flag.taken_action }.from('pending').to('notified_user')
+    end
+
+    it 'sets the flags status to resolved' do
+      expect { service }.to change { flag.status }.from('active').to('resolved')
+    end
+
+    it 'sets the resolved_by_id to the id of the current admin user' do
+      expect { service }.to change { flag.resolved_by_id }.from(nil).to(admin.id)
+    end
+
+    it 'calls the notifications call method' do
+      service
+      expect(notification).to have_received(:call)
```

Comment:
can we assert the flag argument is passed to this service too? its just another little sanity check to ensure we are sending the right thing to it. 

`expect(notification).to have_received(:call).with(flag)`